project/src/main/webapp/gmail.js

```diff
@@ -0,0 +1,365 @@
+//                     'userId': 'me',
+//                     'pageToken': nextPageToken,
+//                     'q': getQuery()
+//                 });
+//                 getPageOfMessages(request, result);
+//             } else {
+//                 for (var i = 0; i < result.length; i++) {
+//                     if (result[i] === undefined) {
+//                         console.log("There are no messages from the query");
+//                     }
+//                     else {
+//                         console.log(result.length);
+//                         console.log(result);
+//                         this.getMessage(result[i].id);
+//                     }
+//                 }
+//             }
+//         });
+//     };
+//     var initialRequest = gapi.client.gmail.users.messages.list({
+//         'userId': 'me',
+//         'q': getQuery()
+//     });
+//     getPageOfMessages(initialRequest, []);
+// }
+
+class gmailAPI {
+    question;
+    emailObjects = {};
+
+    constructor(question) {
+        this.question = question;
+        this.emailObjects = {};
+    }
+
+
+    getEmailBody(decodedEmail) {
+        const emailBodyStartIndex = decodedEmail.indexOf("Content-Type: text/plain; charset=\"UTF-8\"");
+        const emailBodyEndIndex = (decodedEmail.substring(emailBodyStartIndex)).indexOf("Content-Type: text/html; charset=\"UTF-8\"")
+        let emailBodyValue = decodedEmail.substring(emailBodyStartIndex, emailBodyStartIndex + emailBodyEndIndex);
+        if (emailBodyValue.indexOf("Forwarded message") != -1) {
+            emailBodyValue = emailBodyValue.substring(0, emailBodyValue.indexOf("Forwarded message"));
+        }
+        const quotedPrintableIndex = emailBodyValue.indexOf("quoted-printable");
+        if (quotedPrintableIndex != -1) {
+            emailBodyValue = emailBodyValue.substring(quotedPrintableIndex + 16);
+        }
+        return emailBodyValue;
+    }
+
+    getEmailDate(decodedEmail) {
+        const emailDateStartIndex = decodedEmail.indexOf("Date:");
+        const emailDateEndIndex = decodedEmail.substring(emailDateStartIndex).indexOf("Message-ID");
+        return decodedEmail.substring(emailDateStartIndex, emailDateStartIndex + emailDateEndIndex);
+    }
+
+    getMessageID(decodedEmail) {
+        const MessageIDStartIndex = decodedEmail.indexOf("Message-ID:") + 12;
+        const MessageIDEndIndex = decodedEmail.substring(MessageIDStartIndex).indexOf("Subject:") + MessageIDStartIndex;
+        let emailDateValue = decodedEmail.substring(MessageIDStartIndex, MessageIDEndIndex);
+        if (emailDateValue.indexOf("X-Notifications") != -1) {
+            emailDateValue = emailDateValue.substring(0, emailDateValue.indexOf("X-Notifications"));
+        }
+        return emailDateValue;
+    }
+
+    /**
+      * Checks if the email is a valid email
+      * 
+      * Criteria 1: Isn't a Daily Insider message - breaks ML model
```

Comment:
Why is this? Are there just too many of them? If that's the case, maybe instead you could just limit the number of emails sent through the ML model (e.g. only send max 10 of the most recent emails or something)